src/frontend/src/components/SearchPage/SearchPage.jsx

```diff
@@ -26,10 +26,15 @@ const useStyles = makeStyles(() => ({
 const SearchPage = () => {
   const { telescopeUrl } = useSiteMetadata();
   const classes = useStyles();
-
-  const [searchText, setSearchText] = useState('');
+  let [keyword] = useQueryParam('q', StringParam);
+  let [filters] = useQueryParam('f', StringParam);
+  if (!keyword) {
```

Comment:
This shouldn't be necessary.  In the UI rendering below, substitute default values if they are missing.

Reply:
I tried to combine them like this " const [searchText, setSearchText] = useState(`${keyword}`|| '' );" But when keyword is missing, "undefined" is shown in the searchbox instead, not ' '. I don't know why it happens.

Reply:
See my comment above.  We don't need state here, since the URL's query params are the state.